BE crash when sqlancer querys

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
```
--java.lang.AssertionError: SELECT * FROM v17 UNION ALL SELECT * FROM v18 UNION ALL SELECT * FROM v19;
--	at sqlancer.common.query.QueryAdapter.checkException(QueryAdapter.java:98)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:128)
--	at sqlancer.ComparatorHelper.getResultSetFirstColumnAsString(ComparatorHelper.java:54)
--	at sqlancer.ComparatorHelper.getCombinedResultSet(ComparatorHelper.java:138)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPViewOracle.checkViewOnSingleUnion(StarRocksTLPViewOracle.java:68)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPViewOracle.check(StarRocksTLPViewOracle.java:42)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:322)
--	at sqlancer.Main$2.run(Main.java:511)
--	at sqlancer.Main$2.runThread(Main.java:489)
--	at sqlancer.Main$2.run(Main.java:479)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
--Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: rpc failed, host: 172.26.92.182
--	at sun.reflect.GeneratedConstructorAccessor5.newInstance(Unknown Source)
--	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
--	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
--	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
--	at com.mysql.jdbc.Util.getInstance(Util.java:408)
--	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
--	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
--	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2497)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2455)
--	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1369)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:118)
--	... 12 more
---- Time: 2022/05/31 23:29:36
-- Database: v19
-- Database version: 5.1.0
-- seed value: 1654010514542
DROP DATABASE IF EXISTS v19;
CREATE DATABASE v19;
USE v19;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NULL ,c_0_1 DECIMAL(34, 13)  NOT NULL ,c_0_2 CHAR(51) NULL ,c_0_3 BOOLEAN NOT NULL ,c_0_4 CHAR(11) NOT NULL ,c_0_5 DATE NOT NULL ,c_0_6 CHAR(11) NOT NULL ,c_0_7 VARCHAR(11) NOT NULL ,c_0_8 VARCHAR(21) NULL ,c_0_9 DATE NULL ,c_0_10 BOOLEAN NOT NULL , INDEX index1_c_0_2(`c_0_2`) USING BITMAP) DUPLICATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 125 day)) DISTRIBUTED BY HASH (c_0_7,c_0_5,c_0_4,c_0_0,c_0_6,c_0_3,c_0_1,c_0_2) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATE NOT NULL ,c_1_1 CHAR(1) NOT NULL ,c_1_2 VARCHAR(1) NULL ,c_1_3 TINYINT NULL ,c_1_4 SMALLINT NOT NULL ,c_1_5 DECIMAL(17, 15)  NULL ,c_1_6 DATE NULL ,c_1_7 DECIMAL(21, 14)  NULL ,c_1_8 DATETIME NOT NULL ,c_1_9 JSON NOT NULL ) DUPLICATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 292 day)) DISTRIBUTED BY HASH (c_1_4,c_1_3) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t1 VALUES ('2010-01-22', '', '', 5, 10, 9.3E-14, '2010-01-20', 3.906965E-8, '2010-01-29 08:00:14', parse_json('{}')), ('2010-01-14', '', '', 0, 4, 9.3E-14, '2010-01-31', 5.092670E-8, '2010-01-04 08:00:14', parse_json('{"AOLfE7": null, "i": [0.018497836646873322, "z", "N", null, true, 0.9594501058077715, 0.018497836646873322, null, -2066273518], "Qy5VjvcN1": {"45": null, "b": false, "c": -1451310051, "2v": null, "Fioxg3LNb": null, "iP": null, "Mw": true, "SaSmdG": true, "Cr": false, "s": -2066273518, "9p": false, "pWCQrn": null, "Oz3o1": null, "vov7xq": null}}')), ('2010-01-16', '', '', 7, 14, 8E-15, NULL, 1.123546E-8, '2010-01-31 08:00:14', parse_json('{"le9": null, "q": {"0": "H", "7Kw": [false, -1328949220, 0.7142534910979964, "UKpcpflQ", null, 374406704, 0.8007488744777472, "", "AiV7gd98F", true, true, false], "T": -1023318534, "gDC": "jujKJx"}, "D0UV5Pm": {"lM3jEj1": true, "JO3vbR3O": {"k": "DZqz"}, "Xd2uUM": -2066273518, "sVjFy": {"sGhmo": null, "V3TYoS": null, "w8": -1835816884, "ei": 331152181, "l8knE": null, "TO": false, "k": false, "pPdOa38Do": 0.506885324475924}, "u5x": [0.7265171792781941, null, -2066273518, null, null, -1432360522, null, null, null, null, null, true, -1878312170, null, 2117350031], "8934h": 365196622, "z": null, "ZMUlZe": {"c": null, "Nq": true, "D": false, "tHbB7u": 0.5658511976568887, "ao6mAnUo": "s", "O6": null, "79aq": null, "BfjW": null, "q": 0.46499685488563547, "pOtmCsw": null, "klB8Y": "zgN", "Pi": "co", "NaVgapc9": -1.451310051E9, "JD9acZT": null, "y7": 0.4923218264998369}, "2jD": 0.7142534910979964, "UttV8YE": {"A": -401847810, "t8LzaX": null, "5Cp": null, "9A": null, "ro2T0": null, "svCoU": true, "n": true, "R": -1088436089, "aU": 0.018497836646873322, "tFpq6uB": "pHnAUo6", "Hbgb": 0.508313646445024, "rK": null, "9": null, "ruB2egKH": 0.08532128507018222}, "3KFE": ""}, "Sw": "NC0WaU", "h5V": 331152181, "CtMqb": 1103075080}'));
INSERT INTO t0 VALUES ('2010-01-26 08:00:14', 56306464.0123096839352, 'RU', false, 'UY', '2010-01-31', 'gKO', 'Ac', '0', '2010-01-22', true), (NULL, 31447166.2263800530686, 'oHgZEw', true, '', '2010-01-18', 'h', 'LnhY', 'zcC', '2010-01-11', false);
INSERT INTO t0 VALUES ('2010-01-09 08:00:14', 46796388.0927405786612, '', false, 'r', '2010-01-02', 'fMD', 'Xa', 'hO', '2010-01-02', false);
TRUNCATE TABLE t1;
INSERT INTO t0 VALUES ('2010-01-07 08:00:14', 67647520.1570213872607, 't', true, 'v', '2010-01-02', 'C', 'pb3', NULL, '2010-01-02', true);
INSERT INTO t1 VALUES ('2010-01-25', '', '', 5, 2, 3.7E-14, '2010-01-05', 2.255532E-8, '2009-12-31 08:00:14', parse_json('{"sMtON67sF": {"3Qoj": 0.8553753767283838, "R": false, "tauX": 0.12896223072571966, "TrW": 0.7096150669488006, "Yi6I": true, "HCVQ7Dm": null, "n": -317178844}, "5GF": {"BVWik9Wr2": {"9T5P0S": "N", "7X49l3": -38146227, "5E": null, "UOeEuio": true, "R": true, "Bpa": 1679326522, "fz": 1687860388, "GXOPiWQ": null, "w": false, "6AUJ2X7B": null, "yU": false, "Us": 0.6627201603136762, "MvzfHtqpl": 2117350031, "Urkzh": null, "CD73yYEo": 0.6355722516934662}, "FX": [0.9589841670382703, null, "JvIb", 0.7265171792781941], "gzUpRPGj": "", "b": 0.7265171792781941, "4": null, "V": "W", "w": -1502330015, "6OgL4zWc": {}, "8": 0.9480432042889171, "jtExgrG3": {"q": false, "jS91GpfMg": 0.1349681394624721}, "xPIYw5y": 1.721403698E9, "1ro": true}, "t8": true, "fPAzEB": 1.10307508E9, "H": {"TU": 1103075080, "7riUIxkS2": false, "HO": {"t": "", "FfJ": true, "F0v8K": 2117475163}, "m": 1721403698, "r": false, "SXKgj": "", "u": 0.8640297130764131, "ECKtYRvVd": "1MHvvhJB", "n7B": 0.6627201603136762, "DLLNfWEW": {"a": true, "yUpeMfi5": 0.7880342687795364, "f": false, "RY": "PHjpo", "bk": null, "s0Oedjm9j": -447155315, "VA": true, "kdlg9Tz": null, "4B": "", "5c": 0.4413674810731708, "t2P": true, "Znf": false, "wK": -1341335371, "aur": 0.03630059886252923}, "Qj": 0.0106519982271156, "QQ87IAZo": -1.023318534E9, "xU": false, "Z9": [true, 0.03444696914867951, 0.27384200518142365, true, "WFIb", null, true, true, null, 0.6828730444762348, null, false, null, false, true]}, "M59NJwJ": {"WmF": 1103075080, "C": {"b": "", "Ru": "6J2yO0", "gAivn4": true, "pz0UPL": "3", "ze7Rvck": -880838409, "yEBg": 1.10307508E9, "BV6BnEU": 985015518, "Vd": 0.6673237532022718, "8FpL": "xa9B", "Q": "", "3iJ": null, "qTz": true, "8": false, "17Ou": -38146227, "Q6c": -341164089}, "e": null, "2dNXwTG2": 0.2828937791622521, "Q5jC7HD": "1n5I", "x": [579198093, "", "", 0.40481308546296957, null], "69zysUq": true, "0f4QON": {"e0OR": null, "HpX8C": -1140263711, "S": "NtMt", "mAUBaFxv": null, "fRX0j": true, "36E5A": true, "nzQPaj": -740912922, "MF": null, "k4Rwt": null, "1lg9oG": -3.8146227E7, "XR": "eJE", "1MaMPf90J": null}, "HE": [null, -2066273518, false, 0.548941503385313, null, false], "bUJ1l": "UD6Z", "oyY": {"Keba9gqt": false, "IKzVZ8": -1373380357, "9": null, "QcZ8U1": null, "k": null, "Y1fKSUer": false, "Gucr0WY": true, "eLid": null, "OEEh": true, "7DkIODmNE": true, "N": null}, "Yx": 0.8276361055510731}, "nw": {"B9wtlChcJ": {"st": false, "7y6zODL": false, "hD649sT": false, "tpyb": 386732373, "zIi6": "", "g": null, "h": false, "i": null, "h8BkL7v": 1674873455, "Rf7ECOl": null, "bM": "2vH"}, "j": [1973989363], "o": [0.7096150669488006, null, null, -1.7976931348623157E308, true, null], "jCw": "95NM72Tue"}, "tCNK9KJ": [386732373, {"uF": "BO", "CzId": 1687860388, "I0j": null, "F0sI2q": 0.37704883255995036, "6XfEjB": 0.46649010361253807, "3lB79nf": -1088436089, "el": false, "y6": "i", "2Hd": null, "KI": 0.14764233698335694, "az0Z": null, "1wdIyT": 0.7908496617098815}, true, {"p6SDM": true, "m6BUMSc": null, "0jC": null, "m": null, "yNpP": 0.3247307650834236}, {"5u": -1451310051, "3": null, "2kiPuz2": false, "CjZ9s": null}, false, -8.1682669E8], "zD": null, "8KA": -1929202293, "fiOZ066Jz": -447155315}'));
INSERT INTO t0 VALUES ('2010-01-06 08:00:14', 65678988.1637845674369, NULL, false, 'adO9qf2', '2010-01-17', '', 'x1N', NULL, '2010-01-11', true), ('2010-01-17 08:00:14', 56727142.7792703147879, 'yL', false, '', '2010-01-24', 'Tq', 'R', 'oo', '2010-01-30', true), ('2010-01-25 08:00:14', 67742653.3735750766713, 't', true, 'D', '2010-01-12', 'd', '5', 'r81Jgm', '2010-01-24', true);
CREATE VIEW v0 AS SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0;
CREATE VIEW v1 AS SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0 WHERE (NOT (CASE  WHEN false THEN false  WHEN false THEN true END));
CREATE VIEW v2 AS SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0 WHERE (NOT ((NOT (CASE  WHEN false THEN false  WHEN false THEN true END))));
CREATE VIEW v46 AS SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0 WHERE (((NOT (CASE  WHEN false THEN false  WHEN false THEN true END))) IS NULL);
CREATE VIEW v3 AS SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0 WHERE (NOT (CASE  WHEN false THEN false  WHEN false THEN true END)) UNION ALL SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0 WHERE (NOT ((NOT (CASE  WHEN false THEN false  WHEN false THEN true END)))) UNION ALL SELECT t1_3233.c_1_0 FROM  t1 AS t1_3233 LEFT ANTI JOIN (SELECT t0_3232.c_0_0, t0_3232.c_0_1, t0_3232.c_0_2, t0_3232.c_0_3, t0_3232.c_0_4, t0_3232.c_0_5, t0_3232.c_0_6, t0_3232.c_0_7, t0_3232.c_0_8, t0_3232.c_0_9, t0_3232.c_0_10 FROM t0 AS t0_3232 WHERE true) subt0 ON t1_3233.c_1_2 = subt0.c_0_2 AND t1_3233.c_1_2 <= CASE  WHEN false THEN subt0.c_0_2 ELSE subt0.c_0_2 END AND t1_3233.c_1_8 >= subt0.c_0_10 AND t1_3233.c_1_8 < subt0.c_0_10 AND t1_3233.c_1_7 > subt0.c_0_0 AND t1_3233.c_1_7 > subt0.c_0_0 WHERE (((NOT (CASE  WHEN false THEN false  WHEN false THEN true END))) IS NULL);
CREATE VIEW v4 AS WITH with_t_0 as (SELECT v46_3258.c_1_0 FROM v46 AS v46_3258 WHERE false) , with_t_1 as (SELECT v2_3256.c_1_0 FROM v2 AS v2_3256 WHERE true) SELECT subv2.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  (SELECT v2_3256.c_1_0 FROM v2 AS v2_3256 WHERE CAST(v2_3256.c_1_0 AS BOOLEAN ) ) subv2 RIGHT JOIN (SELECT t0_3252.c_0_0, t0_3252.c_0_1, t0_3252.c_0_2, t0_3252.c_0_3, t0_3252.c_0_4, t0_3252.c_0_5, t0_3252.c_0_6, t0_3252.c_0_7, t0_3252.c_0_8, t0_3252.c_0_9, t0_3252.c_0_10 FROM t0 AS t0_3252 WHERE false) subt0 ON subv2.c_1_0 = subt0.c_0_4 AND subv2.c_1_0 = subt0.c_0_4 AND subv2.c_1_0 <= subt0.c_0_3 AND subv2.c_1_0 != BITMAP_CONTAINS(bitmap_hash(''), 1445520523) ,  t1 AS t1_3253 RIGHT SEMI JOIN v46 AS v46_3258 ON t1_3253.c_1_7 = v46_3258.c_1_0 AND t1_3253.c_1_7 < '1970-01-10' AND t1_3253.c_1_8 > v46_3258.c_1_0 AND t1_3253.c_1_8 != v46_3258.c_1_0 AND t1_3253.c_1_1 <= v46_3258.c_1_0 AND t1_3253.c_1_1 > v46_3258.c_1_0,  v3 AS v3_3257 LEFT SEMI JOIN (SELECT v0_3254.c_1_0 FROM v0 AS v0_3254 WHERE (v0_3254.c_1_0) IN ('1970-01-21') ) subv0 ON v3_3257.c_1_0 = subv0.c_1_0 AND '1970-01-04' = subv0.c_1_0 AND v3_3257.c_1_0 > subv0.c_1_0 AND v3_3257.c_1_0 != subv0.c_1_0;
CREATE VIEW v63 AS SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263;
CREATE VIEW v4 AS SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263 WHERE false;
CREATE VIEW v5 AS SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263 WHERE (NOT (false));
CREATE VIEW v6 AS SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263 WHERE ((false) IS NULL);
CREATE VIEW v45 AS SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263 WHERE false UNION ALL SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263 WHERE (NOT (false)) UNION ALL SELECT v3_3264.c_1_0 FROM v3 AS v3_3264, t1 AS t1_3260, v1 AS v1_3262, v0 AS v0_3261, v46 AS v46_3265, t0 AS t0_3259, v2 AS v2_3263 WHERE ((false) IS NULL);
CREATE VIEW v7 AS SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0;
CREATE VIEW v8 AS SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0 WHERE false;
CREATE VIEW v9 AS SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0 WHERE (NOT (false));
CREATE VIEW v10 AS SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0 WHERE ((false) IS NULL);
CREATE VIEW v90 AS SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0 WHERE false UNION ALL SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0 WHERE (NOT (false)) UNION ALL SELECT v6_3314.c_1_0 FROM  v6 AS v6_3314 FULL JOIN (SELECT t1_3305.c_1_0, t1_3305.c_1_1, t1_3305.c_1_2, t1_3305.c_1_3, t1_3305.c_1_4, t1_3305.c_1_5, t1_3305.c_1_6, t1_3305.c_1_7, t1_3305.c_1_8 FROM t1 AS t1_3305) subt1 ON v6_3314.c_1_0 = subt1.c_1_4 AND v6_3314.c_1_0 > ((1720505016)-(805066069)) AND v6_3314.c_1_0 = subt1.c_1_0 AND v6_3314.c_1_0 < subt1.c_1_0,  v4 AS v4_3310 LEFT ANTI JOIN (SELECT v45_3311.c_1_0 FROM v45 AS v45_3311 WHERE ((v45_3311.c_1_0) IS NOT NULL)) subv45 ON v4_3310.c_1_0 = subv45.c_1_0 AND v4_3310.c_1_0 = '1969-12-17' AND v4_3310.c_1_0 != subv45.c_1_0 AND CAST(v4_3310.c_1_0 AS DATE )  > subv45.c_1_0 WHERE ((false) IS NULL);
CREATE VIEW v11 AS SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0;
CREATE VIEW v12 AS SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0 WHERE false;
CREATE VIEW v13 AS SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0 WHERE (NOT (false));
CREATE VIEW v14 AS SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0 WHERE ((false) IS NULL);
CREATE VIEW v15 AS SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0 WHERE false UNION ALL SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0 WHERE (NOT (false)) UNION ALL SELECT v8_3388.c_1_0 FROM v5 AS v5_3384, v90 AS v90_3390, v7 AS v7_3387, v1 AS v1_3377, t1 AS t1_3375, v4 AS v4_3381, v46 AS v46_3383, v9 AS v9_3389, v63 AS v63_3386, v0 AS v0_3376, v45 AS v45_3382, v2 AS v2_3379,  v10 AS v10_3378 RIGHT ANTI JOIN v8 AS v8_3388 ON v10_3378.c_1_0 = v8_3388.c_1_0 AND v10_3378.c_1_0 > v8_3388.c_1_0 AND v10_3378.c_1_0 >= v8_3388.c_1_0 AND v10_3378.c_1_0 < v8_3388.c_1_0 WHERE ((false) IS NULL);
CREATE VIEW v16 AS WITH with_t_0 as (SELECT v3_3480.c_1_0 FROM v3 AS v3_3480) , with_t_1 as (SELECT v0_3471.c_1_0 FROM v0 AS v0_3471 WHERE (CASE  WHEN (v0_3471.c_1_0) IN ('1969-12-22', NULL, '1969-12-25')  THEN v0_3471.c_1_0 ELSE v0_3471.c_1_0 END) IN (NULL, '1970-01-25', '1970-01-12', '1969-12-29') ) SELECT v8_3488.c_1_0 FROM v8 AS v8_3488, v5 AS v5_3484, v0 AS v0_3471, v3 AS v3_3480, v45 AS v45_3482, v6 AS v6_3485, t0 AS t0_3469, v4 AS v4_3481, t1 AS t1_3470, v46 AS v46_3483, with_t_0 AS with_t_0, with_t_1 AS with_t_1;
CREATE VIEW v17 AS WITH with_t_0 as (SELECT v3_3480.c_1_0 FROM v3 AS v3_3480) , with_t_1 as (SELECT v0_3471.c_1_0 FROM v0 AS v0_3471 WHERE (CASE  WHEN (v0_3471.c_1_0) IN ('1969-12-22', NULL, '1969-12-25')  THEN v0_3471.c_1_0 ELSE v0_3471.c_1_0 END) IN (NULL, '1970-01-25', '1970-01-12', '1969-12-29') ) SELECT v8_3488.c_1_0 FROM v8 AS v8_3488, v5 AS v5_3484, v0 AS v0_3471, v3 AS v3_3480, v45 AS v45_3482, v6 AS v6_3485, t0 AS t0_3469, v4 AS v4_3481, t1 AS t1_3470, v46 AS v46_3483, with_t_0 AS with_t_0, with_t_1 AS with_t_1 WHERE (CAST(v45_3482.c_1_0 AS DATE ) )  BETWEEN ('1969-12-29') AND (v3_3480.c_1_0) ;
CREATE VIEW v18 AS WITH with_t_0 as (SELECT v3_3480.c_1_0 FROM v3 AS v3_3480) , with_t_1 as (SELECT v0_3471.c_1_0 FROM v0 AS v0_3471 WHERE (CASE  WHEN (v0_3471.c_1_0) IN ('1969-12-22', NULL, '1969-12-25')  THEN v0_3471.c_1_0 ELSE v0_3471.c_1_0 END) IN (NULL, '1970-01-25', '1970-01-12', '1969-12-29') ) SELECT v8_3488.c_1_0 FROM v8 AS v8_3488, v5 AS v5_3484, v0 AS v0_3471, v3 AS v3_3480, v45 AS v45_3482, v6 AS v6_3485, t0 AS t0_3469, v4 AS v4_3481, t1 AS t1_3470, v46 AS v46_3483, with_t_0 AS with_t_0, with_t_1 AS with_t_1 WHERE (NOT ((CAST(v45_3482.c_1_0 AS DATE ) )  BETWEEN ('1969-12-29') AND (v3_3480.c_1_0) ));
CREATE VIEW v19 AS WITH with_t_0 as (SELECT v3_3480.c_1_0 FROM v3 AS v3_3480) , with_t_1 as (SELECT v0_3471.c_1_0 FROM v0 AS v0_3471 WHERE (CASE  WHEN (v0_3471.c_1_0) IN ('1969-12-22', NULL, '1969-12-25')  THEN v0_3471.c_1_0 ELSE v0_3471.c_1_0 END) IN (NULL, '1970-01-25', '1970-01-12', '1969-12-29') ) SELECT v8_3488.c_1_0 FROM v8 AS v8_3488, v5 AS v5_3484, v0 AS v0_3471, v3 AS v3_3480, v45 AS v45_3482, v6 AS v6_3485, t0 AS t0_3469, v4 AS v4_3481, t1 AS t1_3470, v46 AS v46_3483, with_t_0 AS with_t_0, with_t_1 AS with_t_1 WHERE (((CAST(v45_3482.c_1_0 AS DATE ) )  BETWEEN ('1969-12-29') AND (v3_3480.c_1_0) ) IS NULL);
```

```
mysql> SELECT * FROM v17 UNION ALL SELECT * FROM v18 UNION ALL SELECT * FROM v19;
ERROR 1064 (HY000): rpc failed, host: 172.26.92.183
```

### Expected behavior (Required)

### Real behavior (Required)
be.out:
```
*** Check failure stack trace: ***
    @          0xb2b3aed  google::LogMessage::Fail()
    @          0xb2b5f5f  google::LogMessage::SendToLog()
    @          0xb2b363e  google::LogMessage::Flush()
    @          0xb2b6569  google::LogMessageFatal::~LogMessageFatal()
    @          0x5295a0f  starrocks::vectorized::Chunk::get_column_by_slot_id()
    @          0x996e3cc  starrocks::vectorized::ColumnRef::get_column()
    @          0x996ce4f  starrocks::vectorized::ColumnRef::evaluate()
    @          0x8ea8db8  starrocks::ExprContext::evaluate()
    @          0x8ea8ade  starrocks::ExprContext::evaluate()
    @          0x7ff2576  starrocks::pipeline::ProjectOperator::push_chunk()
    @          0x80ee3b5  starrocks::pipeline::PipelineDriver::process()
    @          0x80bde85  starrocks::pipeline::GlobalDriverExecutor::_worker_thread()
    @          0x80bcbe6  _ZZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiENKUlvE2_clEv
    @          0x80c4988  _ZSt13__invoke_implIvRZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiEUlvE2_JEET_St14__invoke_otherOT0_DpOT1_
    @          0x80c4087  _ZSt10__invoke_rIvRZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiEUlvE2_JEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EES6_E4typeEOS7_DpOS8_
    @          0x80c34b9  _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiEUlvE2_E9_M_invokeERKSt9_Any_data
    @          0x4ac1332  std::function<>::operator()()
    @          0x6632eca  starrocks::FunctionRunnable::run()
    @          0x662f9fe  starrocks::ThreadPool::dispatch_thread()
    @          0x664cff2  std::__invoke_impl<>()
    @          0x664c8d5  std::__invoke<>()
    @          0x664bcec  _ZNSt5_BindIFMN9starrocks10ThreadPoolEFvvEPS1_EE6__callIvJEJLm0EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE
    @          0x664a858  std::_Bind<>::operator()<>()
    @          0x664773e  std::__invoke_impl<>()
    @          0x6645706  _ZSt10__invoke_rIvRSt5_BindIFMN9starrocks10ThreadPoolEFvvEPS2_EEJEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EESA_E4typeEOSB_DpOSC_
    @          0x6640ea1  std::_Function_handler<>::_M_invoke()
    @          0x4ac1332  std::function<>::operator()()
    @          0x6619fd7  starrocks::Thread::supervise_thread()
    @     0x7f6365c4ee65  start_thread
    @     0x7f636526988d  __clone
    @              (nil)  (unknown)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+-------------------+
| current_version() |
+-------------------+
| MAIN-ASAN 54903b2 |
+-------------------+
1 row in set (0.03 sec)

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerTraceUtil.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerTraceUtil.java
@@ -77,12 +77,15 @@ import com.starrocks.sql.optimizer.operator.physical.PhysicalTopNOperator;
 import com.starrocks.sql.optimizer.operator.physical.PhysicalUnionOperator;
 import com.starrocks.sql.optimizer.operator.physical.PhysicalValuesOperator;
 import com.starrocks.sql.optimizer.operator.physical.PhysicalWindowOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
 import com.starrocks.sql.optimizer.rule.Rule;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class OptimizerTraceUtil {
     private static final Logger LOG = LogManager.getLogger(OptimizerTraceUtil.class);
@@ -286,17 +289,20 @@ public class OptimizerTraceUtil {
 
         @Override
         public String visitLogicalUnion(LogicalUnionOperator node, Void context) {
-            return super.visitLogicalUnion(node, context);
+            return getSetOperationBuilder("LogicalUnionOperator", node.getOutputColumnRefOp(),
+                    node.getChildOutputColumns());
         }
 
         @Override
         public String visitLogicalExcept(LogicalExceptOperator node, Void context) {
-            return super.visitLogicalExcept(node, context);
+            return getSetOperationBuilder("LogicalExceptOperator", node.getOutputColumnRefOp(),
+                    node.getChildOutputColumns());
         }
 
         @Override
         public String visitLogicalIntersect(LogicalIntersectOperator node, Void context) {
-            return super.visitLogicalIntersect(node, context);
+            return getSetOperationBuilder("LogicalIntersectOperator", node.getOutputColumnRefOp(),
+                    node.getChildOutputColumns());
         }
 
         @Override
@@ -495,17 +501,36 @@ public class OptimizerTraceUtil {
 
         @Override
         public String visitPhysicalUnion(PhysicalUnionOperator node, Void context) {
-            return super.visitPhysicalUnion(node, context);
+            return getSetOperationBuilder("PhysicalUnionOperator", node.getOutputColumnRefOp(),
+                    node.getChildOutputColumns());
         }
 
         @Override
         public String visitPhysicalExcept(PhysicalExceptOperator node, Void context) {
-            return super.visitPhysicalExcept(node, context);
+            return getSetOperationBuilder("PhysicalExceptOperator", node.getOutputColumnRefOp(),
+                    node.getChildOutputColumns());
         }
 
         @Override
         public String visitPhysicalIntersect(PhysicalIntersectOperator node, Void context) {
-            return super.visitPhysicalIntersect(node, context);
+            return getSetOperationBuilder("PhysicalIntersectOperator", node.getOutputColumnRefOp(),
+                    node.getChildOutputColumns());
+        }
+
+        @NotNull
+        private String getSetOperationBuilder(String name, List<ColumnRefOperator> outputColumnRefOp,
+                                              List<List<ColumnRefOperator>> childOutputColumns) {
+            StringBuilder sb = new StringBuilder(name);
+            sb.append("{");
+            sb.append("output=[").append(outputColumnRefOp.stream().map(ColumnRefOperator::toString)
+                    .collect(Collectors.joining(", "))).append("], ");
+
+            String child = childOutputColumns.stream()
+                    .map(l -> l.stream().map(ColumnRefOperator::toString).collect(Collectors.joining(", ")))
+                    .collect(Collectors.joining(", "));
+            
+            sb.append(child).append("}");
+            return sb.toString();
         }
 
         @Override

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinOrder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinOrder.java
@@ -19,6 +19,7 @@ import com.starrocks.sql.optimizer.operator.logical.LogicalJoinOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.ReplaceColumnRefRewriter;
 import com.starrocks.sql.optimizer.statistics.StatisticsCalculator;
 import com.starrocks.statistic.Constants;
 
@@ -26,8 +27,6 @@ import java.util.BitSet;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.function.Function;
-import java.util.stream.Collectors;
 
 public abstract class JoinOrder {
     /**
@@ -296,16 +295,24 @@ public abstract class JoinOrder {
             return;
         }
 
-        Map<ColumnRefOperator, ScalarOperator> projection = exprInfo.expr.getOutputColumns()
-                .getStream().mapToObj(context.getColumnRefFactory()::getColumnRef)
-                .collect(Collectors.toMap(Function.identity(), Function.identity()));
-        projection.putAll(expression);
+        Map<ColumnRefOperator, ScalarOperator> projection = Maps.newHashMap();
         if (exprInfo.expr.getOp().getProjection() != null) {
             projection.putAll(exprInfo.expr.getOp().getProjection().getColumnRefMap());
+        } else {
+            exprInfo.expr.getOutputColumns().getStream().mapToObj(context.getColumnRefFactory()::getColumnRef)
+                    .forEach(c -> projection.put(c, c));
         }
+
+        // merge two projection, keep same logical from MergeTwoProjectRule
+        ReplaceColumnRefRewriter rewriter = new ReplaceColumnRefRewriter(projection);
+        Map<ColumnRefOperator, ScalarOperator> resultMap = Maps.newHashMap();
+        for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : expression.entrySet()) {
+            resultMap.put(entry.getKey(), rewriter.rewrite(entry.getValue()));
+        }
+
         Operator.Builder builder = OperatorBuilderFactory.build(exprInfo.expr.getOp());
         exprInfo.expr = OptExpression.create(
-                builder.withOperator(exprInfo.expr.getOp()).setProjection(new Projection(projection)).build(),
+                builder.withOperator(exprInfo.expr.getOp()).setProjection(new Projection(resultMap)).build(),
                 exprInfo.expr.getInputs());
         exprInfo.expr.deriveLogicalPropertyItself();
     }

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinOrder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinOrder.java
@@ -309,6 +309,7 @@ public abstract class JoinOrder {
         for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : expression.entrySet()) {
             resultMap.put(entry.getKey(), rewriter.rewrite(entry.getValue()));
         }
+        resultMap.putAll(projection);
 
         Operator.Builder builder = OperatorBuilderFactory.build(exprInfo.expr.getOp());
         exprInfo.expr = OptExpression.create(

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java
@@ -2,7 +2,9 @@
 
 package com.starrocks.sql.optimizer.rule.join;
 
+import com.google.common.base.Preconditions;
 import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
 import com.starrocks.common.FeConstants;
 import com.starrocks.sql.optimizer.ExpressionContext;
 import com.starrocks.sql.optimizer.OptExpression;
@@ -83,7 +85,6 @@ public class ReorderJoinRule extends Rule {
         for (OptExpression joinExpr : reorderTopKResult) {
             ColumnRefSet outputColumns = new ColumnRefSet();
             Map<ColumnRefOperator, ScalarOperator> projectMap = new HashMap<>();
-            Map<ColumnRefOperator, ScalarOperator> commonProjectMap = new HashMap<>();
             if (oldRoot.getProjection() == null) {
                 innerJoinRoot.getInputs().forEach(opt -> outputColumns.union(opt.getOutputColumns()));
 
@@ -93,7 +94,6 @@ public class ReorderJoinRule extends Rule {
             } else {
                 outputColumns.union(oldRoot.getProjection().getOutputColumns());
                 projectMap.putAll(oldRoot.getProjection().getColumnRefMap());
-                commonProjectMap.putAll(oldRoot.getProjection().getCommonSubOperatorMap());
             }
 
             ColumnRefSet newRootInputColumns = new ColumnRefSet();
@@ -118,7 +118,7 @@ public class ReorderJoinRule extends Rule {
                             replaceColumnRefRewriter.rewrite(scalarOperator));
                 }
             }
-            joinExpr.getOp().setProjection(new Projection(projectMap, commonProjectMap));
+            joinExpr.getOp().setProjection(new Projection(projectMap));
             ColumnRefSet requireInputColumns = ((LogicalJoinOperator) joinExpr.getOp()).getRequiredChildInputColumns();
             requireInputColumns.union(outputColumns);
 
@@ -177,23 +177,37 @@ public class ReorderJoinRule extends Rule {
             this.optimizerContext = optimizerContext;
         }
 
-        public OptExpression rewrite(OptExpression optExpression, ColumnRefSet pruneOutputColumns) {
+        public OptExpression rewrite(OptExpression optExpression, ColumnRefSet requiredColumns) {
             Operator operator = optExpression.getOp();
-            if (operator.getProjection() != null) {
+            if (operator.getProjection() != null && operator.getProjection().getColumnRefMap().size() > 1) {
                 Projection projection = operator.getProjection();
 
-                for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : projection.getColumnRefMap().entrySet()) {
-                    if (!entry.getValue().isColumnRef()) {
-                        return optExpression;
+                List<ColumnRefOperator> outputColumns = Lists.newArrayList();
+                for (ColumnRefOperator key : projection.getColumnRefMap().keySet()) {
+                    if (requiredColumns.contains(key)) {
+                        outputColumns.add(key);
+                    }
+                }
+
+                if (outputColumns.size() == 0) {
+                    outputColumns.add(Utils.findSmallestColumnRef(projection.getOutputColumns()));
+                }
+
+                if (outputColumns.size() != projection.getColumnRefMap().size()) {
+                    Map<ColumnRefOperator, ScalarOperator> newOutputProjections = Maps.newHashMap();
+                    for (ColumnRefOperator ref : outputColumns) {
+                        newOutputProjections.put(ref, projection.getColumnRefMap().get(ref));
                     }
 
-                    if (!entry.getKey().equals(entry.getValue())) {
-                        return optExpression;
-                    }
+                    optExpression.getOp().setProjection(new Projection(newOutputProjections));
+                }
+
+                for (ScalarOperator value : optExpression.getOp().getProjection().getColumnRefMap().values()) {
+                    requiredColumns.union(value.getUsedColumns());
                 }
             }
 
-            return optExpression.getOp().accept(this, optExpression, pruneOutputColumns);
+            return optExpression.getOp().accept(this, optExpression, requiredColumns);
         }
 
         @Override
@@ -212,12 +226,8 @@ public class ReorderJoinRule extends Rule {
             }
             requireColumns = ((LogicalJoinOperator) optExpression.getOp()).getRequiredChildInputColumns();
 
-            ColumnRefSet childInputColumns = new ColumnRefSet();
-            optExpression.getInputs().forEach(opt -> childInputColumns.union(
-                    ((LogicalOperator) opt.getOp()).getOutputColumns(new ExpressionContext(opt))));
-
             LogicalJoinOperator joinOperator = (LogicalJoinOperator) optExpression.getOp();
-            if (!newOutputColumns.isEmpty()) {
+            if (joinOperator.getProjection() == null && !newOutputColumns.isEmpty()) {
                 joinOperator = new LogicalJoinOperator.Builder()
                         .withOperator((LogicalJoinOperator) optExpression.getOp())
                         .setProjection(new Projection(newOutputColumns.getStream()
@@ -225,6 +235,9 @@ public class ReorderJoinRule extends Rule {
                                 .collect(Collectors.toMap(Function.identity(), Function.identity())),
                                 new HashMap<>()))
                         .build();
+            } else if (joinOperator.getProjection() != null) {
+                Preconditions.checkState(
+                        joinOperator.getProjection().getColumnRefMap().size() >= newOutputColumns.cardinality());
             }
 
             requireColumns.union(newOutputColumns);

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/ReorderJoinRule.java
@@ -224,7 +224,6 @@ public class ReorderJoinRule extends Rule {
                     newOutputColumns.union(id);
                 }
             }
-            requireColumns = ((LogicalJoinOperator) optExpression.getOp()).getRequiredChildInputColumns();
 
             LogicalJoinOperator joinOperator = (LogicalJoinOperator) optExpression.getOp();
             if (joinOperator.getProjection() == null && !newOutputColumns.isEmpty()) {
@@ -237,9 +236,10 @@ public class ReorderJoinRule extends Rule {
                         .build();
             } else if (joinOperator.getProjection() != null) {
                 Preconditions.checkState(
-                        joinOperator.getProjection().getColumnRefMap().size() >= newOutputColumns.cardinality());
+                        newOutputColumns.cardinality() >= joinOperator.getProjection().getColumnRefMap().size());
             }
 
+            requireColumns = ((LogicalJoinOperator) optExpression.getOp()).getRequiredChildInputColumns();
             requireColumns.union(newOutputColumns);
             OptExpression left = rewrite(optExpression.inputAt(0), (ColumnRefSet) requireColumns.clone());
             OptExpression right = rewrite(optExpression.inputAt(1), (ColumnRefSet) requireColumns.clone());

